pylsp/workspace.py

```diff
@@ -129,6 +129,84 @@ def _create_document(self, doc_uri, source=None, version=None):
         )
 
 
+def get_well_formatted_range(range):
```

Comment:
All functions you added here are not used directly in this file, but only in our tests. So I don't understand why they are here and their usage.

Is not `apply_text_edits` already implemented in the `apply_change` method of the `Document`?

Reply:
`apply_change` handles [TextDocumentContentChangeEvent](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentContentChangeEvent) where as `apply_text_edits` handles [TextEdit](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textEdit). They have different data structure and semantics in the protocol.

Happy to move them to tests, but thought it might be useful to have a function that handles `TextEdit`s. Might be a YAGNI, happy to move to test file.

Reply:
> apply_change handles TextDocumentContentChangeEvent where as apply_text_edits handles TextEdit

Ok, I see. Then please move all these functions to `pylsp/text_edits.py`.

Reply:
You got it